library/ssl_tls.c

```diff
@@ -1626,8 +1708,8 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
         return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
     }
 
-    post_avail = rec->buf_len - ( rec->data_len + rec->data_offset );
     data = rec->buf + rec->data_offset;
+    post_avail = rec->buf_len - ( rec->data_len + rec->data_offset );
```

Comment:
This seems redundant with [line 1764](https://github.com/ARMmbed/mbedtls/pull/2619/commits/d298533c4cc48671b3575a5f3e41b64126c68482#diff-995204bbb15401337db9b64bc314e1f6R1764) below.